code/game/objects/items/tools/shovel_tools.dm

```diff
@@ -200,4 +203,16 @@
 	. = ..()
 	if(sharp)
 		to_chat(user, "<span class='notice'> This one has been sharpened and can no longer be folded.</span>")
+/obj/item/tool/shovel/unique_action(mob/user)
+	entrenchingmode = !entrenchingmode
+	if(entrenchingmode)
+		user.visible_message("<span class='notice'>[user] reconfigures [src] to entrenching mode.</span>")
+	else
+		user.visible_message("<span class='notice'>[user] reconfigures [src] to digging mode.</span>")
 
+/obj/item/tool/shovel/proc/dig_trench(turf/target, mob/user)
```

Comment:
You need to autodoc procs too

Reply:
You are not checking if there is already a trench here. You should also probably check for cades or other obstacles

Reply:
Okay, ill get a check for turrets and the such. I probably wouldnt check for Barricades